tests/integration/clientv3/ordering_util_test.go

```diff
@@ -72,8 +72,13 @@ func TestEndpointSwitchResolvesViolation(t *testing.T) {
 	}
 
 	t.Logf("Reconfigure client to speak only to the 'partitioned' member")
-	cli.SetEndpoints(clus.Members[2].GRPCURL())
-	time.Sleep(1 * time.Second) // give enough time for the operation
+	cfg = clientv3.Config{Endpoints: []string{clus.Members[2].GRPCURL()}}
```

Comment:
I think this might circumvent the problem which is UpdateState[1] is non-blocking. So the SetEndpoints method will return before the client is ready. In testing, I found a wait of even 1 ms had zero failures. I wonder if there is a way we could use something like `WaitForStateChange` (experimental API but something along those lines) to block so that we could remove sleep [2] from all integration tests?

[1] https://github.com/etcd-io/etcd/blob/e2d67f2e3bfa6f72178e26557bb22cc1482c418c/client/v3/internal/resolver/resolver.go#L72
[2] https://github.com/etcd-io/etcd/issues/13451